src/Solnet.Rpc/SolanaRpcBatchComposer.cs

```diff
@@ -0,0 +1,384 @@
+﻿using Solnet.Rpc.Core;
+using Solnet.Rpc.Core.Http;
+using Solnet.Rpc.Messages;
+using Solnet.Rpc.Models;
+using Solnet.Rpc.Types;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Solnet.Rpc
+{
+    /// <summary>
+    /// This object allows a caller to compose a batch of RPC requests for batch submission
+    /// </summary>
+    public class SolanaRpcBatchComposer
+    {
+
+        /// <summary>
+        /// Batch of requests and their handlers
+        /// </summary>
+        private List<RpcBatchReqRespItem> _reqs;
+
+        /// <summary>
+        /// Message Id generator.
+        /// </summary>
+        private readonly IdGenerator _idGenerator = new IdGenerator();
```

Comment:
Maybe should expose the IRpcClient id generator as internal, and re-use it?

Reply:
Good call.
Changing this, but it means a slight adjustment to the the lifecycle of a `SolanaRpcBatchComposer` as it will require a instance of `SolanaRpcClient` on construction to be able to share the `IdGenerator` to compose batches.

Reply:
This rabbit hole required exposing ID generation functionality via `IRpcClient` interface, added `GetNextIdForReq` function for this.